core/src/com/week1/game/Model/GameEngine.java

```diff
@@ -83,13 +89,15 @@ public void receiveMessages(List<? extends GameMessage> messages) {
             adapter.sendMessage(new CheckSyncMessage(enginePlayerId, MessageType.CHECKSYNC, getGameStateHash(), communicationTurn));
 
             // Log the state to the file
-            if (writer != null) {
-                try {
-                    String newContent = "Turn: " + communicationTurn + " hash: " + getGameStateHash() + " String: " + getGameStateString() + "\n";
-                    writer.append(newContent);
-                    writer.flush();
-                } catch (IOException e) {
-                    e.printStackTrace();
+            if (PREFS.getBoolean("doLog")) {
```

Comment:
I would just save this boolean somewhere to you don't have to read from prefs every time

Reply:
Saved it in StatsConfig